source/codegen/templates/proto_helpers.mako

```diff
@@ -79,23 +85,38 @@ message ${common_helpers.snake_to_pascal(function)}Request {
 % for parameter in input_parameters:
 <%
   parameter_name = common_helpers.camel_to_snake(parameter["name"])
-  grpc_field_number = proto_helpers.generate_parameter_field_number(parameter, used_indexes)
+  is_array = common_helpers.is_array(parameter["type"])
   if 'grpc_type' in parameter:
     parameter_type = parameter['grpc_type']
   else:
-    parameter_type = proto_helpers.determine_function_parameter_type(parameter, service_class_prefix)
+    parameter_type = proto_helpers.get_grpc_type_from_ivi(parameter["type"], is_array, service_class_prefix)
```

Comment:
Is this replacement of function `determine_function_parameter_type` with `get_grpc_type_from_ivi` expected?

Reply:
Yes, I have removed `determine_function_parameter_type` function. That function was used for determining parameter type for enum parameter as well as other parameters. For other parameters, it used to call `get_grpc_type_from_ivi`. That function doesn't make sense anymore, as even with enum we could have two potential types, one for Mapped enum and other for Unampped enum. It is difficult to customize `determine_function_parameter_type` to handle 2 different enum types, so I have inlined type generation for enums.